src/components/ImageEditing/ImageEditingContainer.tsx

```diff
@@ -0,0 +1,243 @@
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  Flex,
+  Box,
+  Button,
+  Text,
+  Input,
+  Center,
+  Slider,
+  SliderTrack,
+  SliderFilledTrack,
+  SliderThumb,
+  useToast,
+} from "@chakra-ui/react";
+import Cropper from "react-easy-crop";
+import { Point, Area } from "react-easy-crop/types";
+import { toBase64, upLoadScreenshot, inputImageCheck } from "@/utils/manageExternalImage";
+import { getCroppedImg } from "@/utils/canvasUtils";
+import { useDispatch } from "react-redux";
+import { updateUserInfo } from "@/store/reducer/userSlice";
+import { updateUser } from "@/components/Login/helpers/userRequests";
+import { useStoreSelector } from "@/store/hooks";
+
+interface Props {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const ImageEditingContainer = ({ isOpen, onClose }: Props) => {
+  const dispatch = useDispatch();
+  const toast = useToast();
+  const userInfo = useStoreSelector((state) => state.user);
+  const [picture, setPicture] = useState<string | null>(null);
+  const [crop, setCrop] = useState<Point>({ x: 0, y: 0 });
+  const [zoom, setZoom] = useState(1);
+  const [rotation, setRotation] = useState(0);
+  const [croppedAreaPixels, setCroppedAreaPixels] = useState<Area | null>(null);
+
+  const onCropComplete = useCallback((croppedArea: Area, croppedAreaPixels: Area) => {
+    setCroppedAreaPixels(croppedAreaPixels);
+  }, []);
+
+  const closeWindow = () => {
+    onClose();
+  };
+
+  const handleInputImage = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files?.length === 0 || !e.target.files) return;
+    if (!inputImageCheck(e.target.files[0], 0, 300, toast)) return;
+    const imgUrl = (await toBase64(e.target.files[0])) as string;
+    setPicture(imgUrl);
+  };
+
+  const handleDragOver = (e: React.DragEvent<HTMLInputElement>) => {
+    e.stopPropagation();
+    e.preventDefault();
+    e.dataTransfer.dropEffect = "copy";
+  };
+
+  const handleDragImage = async (e: React.DragEvent<HTMLInputElement>) => {
+    e.stopPropagation();
+    e.preventDefault();
+    const file = e.dataTransfer.files[0];
+    if (!inputImageCheck(file, 0, 300, toast)) return;
+    const imgUrl = (await toBase64(file)) as string;
+    setPicture(imgUrl);
+  };
+
+  const handleReset = () => {
+    setZoom(1);
+    setRotation(0);
+  };
+
+  const handleApply = useCallback(async () => {
+    if (!picture || !croppedAreaPixels) return;
+    const croppedImage = (await getCroppedImg(picture, croppedAreaPixels, rotation)) as string;
+    const uploadedImageUrl = await upLoadScreenshot(
+      croppedImage,
+      toast,
+      process.env.NEXT_PUBLIC_AVATAR_ALBUM_NAME
+    );
+    localStorage.setItem(
+      "UserInfo",
+      JSON.stringify({ ...userInfo, profileImgUrl: uploadedImageUrl })
+    );
+    dispatch(updateUserInfo({ ...userInfo, profileImgUrl: uploadedImageUrl }));
+    updateUser({ ...userInfo, profileImgUrl: uploadedImageUrl });
+    closeWindow();
+  }, [picture, croppedAreaPixels, rotation]);
```

Comment:
You have a lot of user image specific logic here inside of this component but this file calls ImageEditingContainer? Or should it name userAvatarEditingContainer? If you want to reuse this component to fulfill any future needs (such as the image upload component for court), I believe this may need a restructure.